### PR TITLE
SALTO-6788-FIX: Add CV for business holiday creation dates

### PR DESCRIPTION
**Pull request by AlmogMesilaty**
SALTO-6788-FIX: Add CV for business holiday creation dates
---

_Additional context for reviewer_

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/zendesk-adapter/src/change_validators/business_hours_schedule_holiday.ts [hunk 1]
```diff
@@ -41,12 +41,16 @@ const isDateRangeExceedsLimit = (instance: InstanceElement): boolean => {
     return false
   }
   const startDate = new Date(value.start_date)
+  startDate.setHours(0, 0, 0, 0)
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/change_validators/business_hours_schedule_holiday.ts:44]: how does this solve the bug that you had?

File: packages/zendesk-adapter/test/change_validators/business_hours_schedule_holiday.test.ts [hunk 2]
```diff
@@ -21,19 +22,21 @@ describe('businessHoursScheduleHolidayChangeValidator', () => {
     })
 
   const dateYearsAgo = (years: number): string => {
-    const date = new Date()
     date.setFullYear(date.getFullYear() - years)
-    return date.toISOString()
+    const ret = date.toISOString()
+    date.setFullYear(date.getFullYear() + years)
```
Comment (edenhassid) [inline packages/zendesk-adapter/test/change_validators/business_hours_schedule_holiday.test.ts:27]:
```suggestion
    date.setFullYear(date.getFullYear()  - years)
```
as the name of the function is dateYearsAgo

File: packages/zendesk-adapter/src/change_validators/business_hours_schedule_holiday.ts (hunk 1 above)
Comment (edenhassid) [reply packages/zendesk-adapter/src/change_validators/business_hours_schedule_holiday.ts:44]: please also write a comment explaining why you are doing this

File: packages/zendesk-adapter/test/change_validators/business_hours_schedule_holiday.test.ts [hunk 3]
```diff
@@ -21,19 +22,21 @@ describe('businessHoursScheduleHolidayChangeValidator', () => {
     })
 
   const dateYearsAgo = (years: number): string => {
-    const date = new Date()
     date.setFullYear(date.getFullYear() - years)
-    return date.toISOString()
+    const ret = date.toISOString()
```
Comment (edenhassid) [inline packages/zendesk-adapter/test/change_validators/business_hours_schedule_holiday.test.ts:26]: why does this solve the issue?

File: packages/zendesk-adapter/test/change_validators/business_hours_schedule_holiday.test.ts [hunk 4]
```diff
@@ -42,33 +45,30 @@ describe('businessHoursScheduleHolidayChangeValidator', () => {
...
-    const holiday = createHolidayInstance('test_holiday', new Date().toISOString(), dateYearsFromNow(2))
+    const holiday = createHolidayInstance('test_holiday', date.toISOString(), dateYearsFromNow(2))
     changes = [toChange({ after: holiday })]
-
     const errors = await businessHoursScheduleHolidayChangeValidator(changes)
     expect(errors).toHaveLength(1)
   })
   it('should error when start date is more than 2 years sooner', async () => {
-    const holiday = createHolidayInstance('test_holiday', dateYearsAgo(3), new Date().toISOString())
+    const holiday = createHolidayInstance('test_holiday', dateYearsAgo(2), date.toISOString())
```
Comment (edenhassid) [inline packages/zendesk-adapter/test/change_validators/business_hours_schedule_holiday.test.ts:59]: why did you change to 2?

**Review by edenhassid — COMMENTED** [review]
nice catch! added some comments :)

File: packages/zendesk-adapter/src/change_validators/business_hours_schedule_holiday.ts (hunk 1 above)
Comment (AlmogMesilaty) [reply packages/zendesk-adapter/src/change_validators/business_hours_schedule_holiday.ts:44]: I am doing this to normalize the date to day resolution only, ignoring time, this is to match the resolution in Zendesk:
image
It solves the bug because similar to the test if someone manually edit a nacl and inserts date two years in the past but earlier than the current time, my old check would have failed.

File: packages/zendesk-adapter/test/change_validators/business_hours_schedule_holiday.test.ts (hunk 3 above)
Comment (AlmogMesilaty) [reply packages/zendesk-adapter/test/change_validators/business_hours_schedule_holiday.test.ts:26]: When I used new Date() multiple time in the tests it created a problem of time difference between them, now I am using it once and create multiple date instances from its value, would not effect the new implementation that look at day-level resolution, but still decided to change.

File: packages/zendesk-adapter/test/change_validators/business_hours_schedule_holiday.test.ts (hunk 4 above)
Comment (AlmogMesilaty) [reply packages/zendesk-adapter/test/change_validators/business_hours_schedule_holiday.test.ts:59]: my mistake, fixed it :)

**Review by edenhassid — APPROVED** [review]
LGTM, thanks!
